### PR TITLE
Change base image to Debian 11

### PR DESCRIPTION
**Pull request by kaidokert**
This updates base runner images to Debian11.

b/312772759

### CODE REVIEW

File: starboard/linux/shared/gyp_configuration.py [hunk 1]
```diff
@@ -39,4 +39,6 @@ def GetTestEnvVariables(self):
     # When failing, 'LeakSanitizer has encountered a fatal error' message would
     # be printed at test shutdown, and env var LSAN_OPTIONS=verbosity=2 would
     # further point to 'Scanning DTLS range ..' prior to crash.
-    return {'ASAN_OPTIONS': 'intercept_tls_get_addr=0'}
+    # quarantine_size is temporarily set to a huge value to prevent recycling
+    # from happening, as it crashes on Debian11+Mesa. TODO: b/321746394
+    return {'ASAN_OPTIONS': 'intercept_tls_get_addr=0:quarantine_size_mb=32768'}
```
Comment (andrewsavage1) [inline starboard/linux/shared/gyp_configuration.py:44]: Does setting this to such a large value have any implications on running tests/cobalt or not?

Comment (kaidokert) [reply starboard/linux/shared/gyp_configuration.py:44]: No implications that i could see so far. The only issue is workers actually need to have that amount of RAM for now, and they do.

File: starboard/tools/api_leak_detector/stub/debug/gn_built_docker_debian11_manifest
Comment (andrewsavage1) [inline starboard/tools/api_leak_detector/stub/debug/gn_built_docker_debian11_manifest:1]: Is there an internal workflow that depends on these files? If so, we should just create a new debian11 file and delete the debian10 one in a follow up to not break internal workflows

Comment (kaidokert) [reply starboard/tools/api_leak_detector/stub/debug/gn_built_docker_debian11_manifest:1]: That's what i initially did yes, but there doesn't appear to be an internal workflow and there probably shouldn't be too - no reason to check this twice.

**Comment by kaidokert** [comment]
I believe this can go in, but then still need to re-land go/cobalt-cl/260221

**Pull request merged**